### PR TITLE
Fix setup guide with missing steps and more info

### PR DESCRIPTION
**Pull request by marcmerlin**
1) Add info for installing renode
2) fix non working autolink for installing vivado
3) add python virtualenv info
4) add required nmigen
5) better instructions on how to activate the code after make load

### CODE REVIEW

File: docs/source/setup-guide.rst [hunk 1]
```diff
@@ -18,36 +18,42 @@ but is significantly more expensive.
 .. _`Mouser`: https://au.mouser.com/ProductDetail/Digilent/410-319?qs=%2Fha2pyFaduiP6GD6DfdhNp6rR4rT1KTVOohSnRQ%252BMgra5hr4M7aEiQ%3D%3D 
 
 
-Step 2: Install Vivado
-----------------------
+Step 2: Install Vivado and renode
+---------------------------------
 
-(can take ~8 hours to download software)
+1. vivaldo
```
Comment (alanvgreen) [inline docs/source/setup-guide.rst:24]: nit: vivado

File: docs/source/setup-guide.rst [hunk 2]
```diff
@@ -80,21 +91,49 @@ as a useful "minimal" system.
...
    $ make clean
 
    # Program the bitstream onto the board. The first run will take several minutes
    # as Vivado synthesizes a bitstream
+   # If this works fine, you will get a chasing LED pattern with the 4 LEDs
    $ make prog
 
    # Build the RISCV program and load it onto the board
-   # Also starts a terminal
+   # Also starts a terminal ( exit the terminal with CTRL+\ )
```
Comment (alanvgreen) [inline docs/source/setup-guide.rst:105]: I didn't know about Ctrl+\ - I have been mashing ctrl-c. 

Hmm. I just tried Ctrl+\ and found it left my terminal in a bad state. Maybe we should recommend mashing ctrl-C? WDYT?

**Review by alanvgreen — APPROVED** [review]
I'm happy for this to go in as-is. Two nits, which I'll leave up to you as to whether to fix or not.

File: docs/source/setup-guide.rst (hunk 1 above)
Comment (marcmerlin) [reply docs/source/setup-guide.rst]: oops, fixed, thanks.

File: docs/source/setup-guide.rst (hunk 2 above)
Comment (marcmerlin) [reply docs/source/setup-guide.rst]: CTRL+C is better actually, I changed it.

**Comment by marcmerlin** [comment]
Thanks for the review @alanvgreen  , PTAL

**Comment by marcmerlin** [comment]
@alanvgreen  because of the perms, you'll need to re-approve and submit, I don't have perms :)

**Comment by marcmerlin** [comment]
Hi @mithro , looks like the approve from @alanvgreen isn't enough to allow me to submit. Are you able to give me an approve that allows submit or submit for me ?

File: docs/source/setup-guide.rst (hunk 2 above)
Comment (tcal-x) [reply docs/source/setup-guide.rst]: The trick is that you need *two* Ctrl-C's to exit litex_term, which may be why Marc originally resorted to Ctrl+\ .    (I'm just catching up on stuff now after returning from vacation.)

**Pull request merged**